joblib/test/test_parallel.py

```diff
@@ -1334,14 +1376,18 @@ def test_multiple_generator_call_separated(backend, n_jobs):
     ('threading', False),
     ('sequential', False),
 ])
-def test_multiple_generator_call_separated_gc(backend, error):
+@parametrize('return_as_1', ["generator", "generator_unordered"])
+@parametrize('return_as_2', ["generator", "generator_unordered"])
```

Comment:
`return_as_outer` / `return_as_inner` would be more explicit.

Reply:
Agree that `return_as_1/2` does not read well but `inner` and `outer` is used in other tests to name levels in nested parallel calls, but those tests only consists in two sequential calls.

Maybe adding this grid of parameters is overkill ? we could just set the same `return_as` for all calls ?

Reply:
Indeed I was confused by the indentation levels. Let's ignore this comment.